internal/service/grid/distributionschedule_resource.go

```diff
@@ -0,0 +1,263 @@
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	niosclient "github.com/infobloxopen/infoblox-nios-go-client/client"
+
+	"github.com/infobloxopen/terraform-provider-nios/internal/flex"
+	"github.com/infobloxopen/terraform-provider-nios/internal/utils"
+)
+
+var readableAttributesForDistributionschedule = "active,start_time,time_zone,upgrade_groups"
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ resource.Resource = &DistributionscheduleResource{}
+var _ resource.ResourceWithImportState = &DistributionscheduleResource{}
+
+func NewDistributionscheduleResource() resource.Resource {
+	return &DistributionscheduleResource{}
+}
+
+// DistributionscheduleResource defines the resource implementation.
+type DistributionscheduleResource struct {
+	client *niosclient.APIClient
+}
+
+func (r *DistributionscheduleResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_" + "grid_distributionschedule"
+}
+
+func (r *DistributionscheduleResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Manages a Distribution Schedule resource within the NIOS Grid.",
+		Attributes:          DistributionscheduleResourceSchemaAttributes,
+	}
+}
+
+func (r *DistributionscheduleResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*niosclient.APIClient)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *niosclient.APIClient, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+func (r *DistributionscheduleResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var (
+		data       DistributionscheduleModel
+		dataGroups []DistributionscheduleUpgradeGroupsModel
+	)
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	listResp, _, err := r.client.GridAPI.
+		DistributionscheduleAPI.
+		List(ctx).
+		ReturnAsObject(1).
+		ReturnFieldsPlus(readableAttributesForDistributionschedule).
+		Execute()
+	if err != nil {
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to list DistributionSchedule: %s", err))
+		return
+	}
+
+	list := listResp.ListDistributionscheduleResponseObject.GetResult()
+
+	if len(list) == 0 {
+		resp.Diagnostics.AddError("Not Found", "No DistributionSchedule object exists in this Grid")
+		return
+	}
+
+	// Extract the singleton ref
+	ref := list[0].GetRef()
```

Comment:
```suggestion
	ref := listObject.Ref
```
Similar change can be made for all other use cases

Reply:
Also if ref is used only once , declaration variable declaration seems unnecessary !